### PR TITLE
Raster transparency API improvements + cleanups

### PR DESCRIPTION
**Pull request by nyalldawson**
- Make classes less like to be created with uninitialized members
- Use opacity instead of transparency, to match API from elsewhere in QGIS
- Add equality operators and Python repr
- Add explicit setting for includeMinimum/Maximum to QgsRasterTransparency.TransparentSingleValuePixel. Allows (via
API only) control over whether the extremities of the range should be included in the transparency.
- Add more unit tests for these classes

No user facing changes here!

### CODE REVIEW

File: src/core/raster/qgsrastertransparency.cpp [hunk 1]
```diff
@@ -97,7 +87,7 @@ int QgsRasterTransparency::alphaValue( double value, int globalTransparency ) co
   //if a match was found use the stored transparency percentage
   if ( myTransparentPixelFound )
   {
-    return static_cast< int >( static_cast< float >( globalTransparency ) * ( 1.0 - ( myTransparentPixel.percentTransparent / 100.0 ) ) );
+    return static_cast< int >( static_cast< float >( globalTransparency ) * myTransparentPixel.opacity );
```
Comment (rouault) [inline src/core/raster/qgsrastertransparency.cpp:90]: a transparency multiplied by an opacity... That gives me some headaches :-)  Is globalTransparency properly named or is it a globalOpacity ?

File: src/core/raster/qgsrastertransparency.cpp [hunk 2]
```diff
@@ -133,7 +123,7 @@ int QgsRasterTransparency::alphaValue( double redValue, double greenValue, doubl
...
```
Comment (rouault) [inline src/core/raster/qgsrastertransparency.cpp:126]: same remark as above

File: src/core/raster/qgsrastertransparency.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/raster/qgsrastertransparency.cpp:90]: It's actually an alpha value :rofl: (0-255, integer). Let me fix this too!

Comment (nyalldawson) [reply src/core/raster/qgsrastertransparency.cpp]: Ok, should be fixed by [729c814](https://github.com/qgis/QGIS/pull/56598/commits/729c81431151efd1e241a3e253a548eaa29fb454)

**Review by rouault — APPROVED** [review]

**Pull request merged**